addons/mail/static/src/js/mail_mention.js

```diff
@@ -0,0 +1,224 @@
+odoo.define('mail.mention', function (require) {
+"use strict";
+
+var core = require('web.core');
+var Model = require('web.Model');
+var Widget = require('web.Widget');
+var form_common = require('web.form_common');
+
+var _t = core._t;
+var QWeb = core.qweb;
+
+/**
+ * ------------------------------------------------------------
+ * Mention widget
+ * ------------------------------------------------------------
+ *
+ * This widget handles @ mention functionality.
+ * Open dropdown when user mention partner with @
+ * Dropdown open with highlighted string
+ * Start search if word have minimum length is 3
+ * Wait for at least 400 milliseconds without typing any new letter
+ * Maximum number of partner in dropdown list is 8
+ */
+var Mention = Widget.extend({
+    template: null,
+    init: function(parent) {
+        this._super.apply(this, arguments);
+        this.$textarea = parent.$el.find(".field_text");
+        this.$dropdown = $("<div class='mention'><ul></ul></div>");
+        this.model_res_partner = new Model("res.partner");
+        this.selected_partners = {};
+        this.search_keys = {};
+        this.partners = {};
+        this.keyup_list = [40, 38, 13, 35, 33, 34, 27];
+        this.partner_limit = 8;
```

Comment:
`partner_limit`, `typing_speed` and `min_charactor` should be options, given as parameters, and the values you give as default options values, completed with `_.defaults()`